golang/main.go

```diff
@@ -0,0 +1,407 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	yaml "gopkg.in/yaml.v2"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func main() {
+
+	// getting config
+	cfg, err := readConfig("app.yaml")
+	if err != nil {
+		printError("unable to read config file", err)
+	}
+
+	fmt.Print(
+		"Please select a communication protocol\n" +
+			"1) HTTP endpoint\n" +
+			"2) PostgreSQL wire protocol\n")
+
+	reader := bufio.NewReader(os.Stdin)
```

Comment:
This looks like it would prompt the user to enter "1" or "2", to define the mode of operation? It would be sweet to use a command line option instead, like `--http` vs. `--postgresql`.

In this manner, the application could be started unattended, which makes testing a bit easier. I think it would be good to make the change right now, even if the code is not accompanied by corresponding software tests yet.

Reply:
applied